src/ui/input/input.tsx

```diff
@@ -0,0 +1,148 @@
+import { ReactNode, ChangeEvent, useState} from "react"
+import { Icon } from "../icon/Icon"
+import style from "./input.module.scss"
+import cnBind from 'classnames/bind'
+
+const cx = cnBind.bind(style)
+
+interface InputProps {
+    className?: string
+    placeholder?: string
+    type?: 'text' | 'password' | 'textarea' | 'number'
+    disabled?: boolean
+    prefixIcon?: ReactNode
+    postfixIcon?: ReactNode
+    passwordHideIcon?: ReactNode
+    passwordShowIcon?: ReactNode
+    iconClear?: ReactNode
+    clearable?: boolean
+    size?: 'md' | 'sm' | 'xs'
+    value?: string | number
+    prepend?: string | ReactNode
+    append?: string | ReactNode
+    rows?: number
+    onChange: (value: string) => void
+}
+interface IconProps{
+    prefix?: ReactNode
+    postfix?: ReactNode
+}
+
+interface ContentProps{
+    prepend?: ReactNode
+    append?: ReactNode
+}
+
+export const Input = ({
+        size="md",
+        className="",
+        placeholder="",
+        type="text",
+        disabled=false,
+        clearable=false, 
+        rows=4,
+        postfixIcon,
+        prefixIcon,
+        passwordHideIcon,
+        passwordShowIcon,
+        iconClear,
+        value,
+        onChange,
+        prepend,
+        append,
+    }: InputProps) => {
+    
+    const [passwordShown, setPasswordShown] = useState(false);
+    
+    const onValueChange = (event: ChangeEvent<HTMLInputElement>) => {
+        onChange(event.target.value)
+    }
+    const inputType = passwordShown ? "text" : type
```

Comment:
Отступы ставь везде...
```suggestion
    }
    
    const inputType = passwordShown ? "text" : type
```

Reply:
исправлено